packages/tooling/fast-tooling/src/data-utilities/shortcut-action.ts

```diff
@@ -0,0 +1,107 @@
+export interface MetaKey {
+    metaKey: true;
+}
+
+export interface ShiftKey {
+    shiftKey: true;
+}
+
+export interface CtrlKey {
+    ctrlKey: true;
+}
+
+export interface AltKey {
+    altKey: true;
+}
+
+export type ModifierKey = AltKey | CtrlKey | ShiftKey | MetaKey;
+
+export interface SpecificKey {
+    /**
+     * The key string value
+     */
+    value: string;
+}
+
+export type KeyConfig = SpecificKey | ModifierKey;
```

Comment:
With this typing, a `KeyConfig` with only a modifier key is valid - is that accurate?

```
const foo: KeyConfig = { // Totally cool
    shiftKey: true
}
```

Reply:
Yes, that's correct.

Reply:
Whoops, so it seems like I misinterpreted what `|` does, I am re-doing the typescript to use exclusionary interfaces so that they can't be combined.